Sources/Commands/SwiftBuildTool.swift

```diff
@@ -157,8 +159,23 @@ public struct SwiftBuildTool: SwiftCommand {
     public init() {}
 }
 
-extension Basics.Diagnostic {
-    static func mutuallyExclusiveArgumentsError(arguments: [String]) -> Self {
-        .error(arguments.map{ "'\($0)'" }.spm_localizedJoin(type: .conjunction) + " are mutually exclusive")
+extension SwiftCommand {
+    public func buildSystemProvider(_ swiftTool: SwiftTool) throws -> BuildSystemProvider {
+        return .init(providers: try swiftTool.defaultBuildSystemProvider.providers.merging([
+            .xcode: { (explicitProduct: String?, cacheBuildManifest: Bool, customBuildParameters: BuildParameters?, customPackageGraphLoader: (() throws -> PackageGraph)?, customOutputStream: OutputByteStream?, customLogLevel: Basics.Diagnostic.Severity?, customObservabilityScope: ObservabilityScope?) throws -> BuildSystem in
+                let graphLoader = { try swiftTool.loadPackageGraph(explicitProduct: explicitProduct) }
+                // FIXME: Implement the custom build command provider also.
```

Comment:
still to do?

Reply:
I just carried this over from the original code, honestly not sure what it means, so maybe we can remove it?

Reply:
Removing this now.